net/src/headers.rs

```diff
@@ -1372,4 +1378,290 @@ mod test {
+        }
+
+        pub(super) fn ipv4_tcp() -> Headers {
+            let mut headers = HeadersBuilder::default();
+            let mut ipv4 = ipv4(NextHeader::TCP);
+            let tcp = tcp();
+            ipv4.set_payload_len(tcp.size().get()).unwrap();
+            headers
+                .eth(Some(eth(EthType::IPV4)))
+                .net(Some(Net::Ipv4(ipv4)))
+                .transport(Some(Transport::Tcp(tcp)))
+                .build()
+                .unwrap()
+        }
+
+        pub(super) fn ipv4_icmp() -> Headers {
+            let mut headers = HeadersBuilder::default();
+            headers
+                .eth(Some(eth(EthType::IPV4)))
+                .net(Some(Net::Ipv4(ipv4(NextHeader::ICMP))))
+                .transport(Some(Transport::Icmp4(icmp4())))
+                .build()
+                .unwrap()
+        }
+
+        pub(super) fn ipv4_udp() -> Headers {
+            let mut headers = HeadersBuilder::default();
+            let mut ipv6 = ipv6(NextHeader::UDP);
+            let udp = udp();
+            ipv6.set_payload_length(udp.size().get());
+            headers
+                .eth(Some(eth(EthType::IPV4)))
+                .net(Some(Net::Ipv6(ipv6)))
+                .transport(Some(Transport::Udp(udp)))
+                .build()
+                .unwrap()
+        }
+
+        pub(super) fn ipv6_tcp() -> Headers {
+            let mut headers = HeadersBuilder::default();
+            let tcp = tcp();
+            let mut ipv6 = ipv6(NextHeader::TCP);
+            ipv6.set_payload_length(tcp.size().get());
+            headers
+                .eth(Some(eth(EthType::IPV6)))
+                .net(Some(Net::Ipv6(ipv6)))
+                .transport(Some(Transport::Tcp(tcp)))
+                .build()
+                .unwrap()
+        }
+
+        pub(super) fn ipv6_udp() -> Headers {
+            let mut headers = HeadersBuilder::default();
+            let udp = udp();
+            let mut ipv6 = ipv6(NextHeader::UDP);
+            ipv6.set_payload_length(udp.size().get());
+            headers
+                .eth(Some(eth(EthType::IPV6)))
+                .net(Some(Net::Ipv6(ipv6)))
+                .transport(Some(Transport::Udp(udp)))
+                .build()
+                .unwrap()
+        }
+
+        pub(super) fn ipv6_icmp() -> Headers {
+            let mut headers = HeadersBuilder::default();
+            let icmp = icmp6();
+            let mut ipv6 = ipv6(NextHeader::ICMP6);
+            ipv6.set_payload_length(icmp.size().get());
+            headers
+                .eth(Some(eth(EthType::IPV6)))
+                .net(Some(Net::Ipv6(ipv6)))
+                .transport(Some(Transport::Icmp6(icmp)))
+                .build()
+                .unwrap()
+        }
+    }
+
+    fn test_checksum(mut headers: Headers) {
+        match &headers.transport {
+            None => {}
+            Some(Transport::Udp(transport)) => {
+                let net = headers.net.clone().unwrap();
```

Comment:
Can we have some tests with a non-zero payload perhaps?  Even if you look at wireshark, find the correct check sum and just hardcode the constant in at least some test cases?

Reply:
Done.